subsys/bootloader/mcuboot_image_header_parser/mcuboot_image_header_parser.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr/types.h>
+#include <string.h>
+#include <image.h>
+#include <errno.h>
+
+/**@brief Reads MCUBoots Image header
+ *
+ * @param header_addr Address of where the image header is located.
+ * @param[out] image_header Pointer to the image header struct which should be
+ * populated.
+ *
+ * @retval 0 If successfully read the IMAGE_HEADER magic meaning an MCUBoot
+ * header was present.
+ * @retval -EINVAL If the image_header magic doesn't match meaning an MCUBoot
+ * header was not found at the given address.
+ */
+static int mcuboot_read_img_hdr(u32_t *header_addr,
```

Comment:
To save the user for some reading, this function could be inlined in `mcuboot_read_version_number`. 

That way we don't need this much docs, and the code will still be clean IMO.

Reply:
Sure, the idea here was to maybe have it as a helper function later if more functionality is added, can move it into `read_version_number`.